config/esp32/components/chip/Kconfig

```diff
@@ -163,6 +163,16 @@ menu "CHIP Core"
             help
                 Option to enable/disable CHIP data model.
 
+        config CHIP_SYSTEM_CONFIG_POOL_USE_HEAP
+            bool "Use heap memory to allocate object pools"
+            default n
+            help
+                This option enables the use of heap memory to allocate object pools.
+                When this option is enabled, it does not pre-allocate the object pools.
+                Additionally, there is no limit, (i.e. free heap is the limit) on the
+                maximum number of entries that can be allocated.
+                This option can be usefull if you run into static DRAM overflow.
```

Comment:
Note that:

1. This option can turn a failure to compile into just a device that does not properly implement the spec (and might fail cert, or worse yet pass cert and then fail to work with actual controllers).  This needs to be called out clearly.
2. The fact that there is no cap on pool sizes may lead to problems in places where embedded code currently assumes the pool size _is_ in fact limited and has no other limiters on the size of things.

Reply:
I have added these points as a NOTE in the config options help.

Reply:
@bzbarsky-apple can you please review the text again!